bundles/admin/admin-announcements/view/AnnouncementsListHandler.js

```diff
@@ -0,0 +1,156 @@
+
+/*
+Handler for admin-announcements forms.
+*/
+
+const getMessage = (key, args) => <Message messageKey={key} messageArgs={args} bundleKey='admin-announcements' />;
+
+/**
+* Remove item from a localStorage() array
+* @param {String} name  The localStorage() key
+* @param {String} value The localStorage() value
+*/
+const removeFromLocalStorageArray = (name, value) => {
+    // Get the existing data
+    var existing = localStorage.getItem(name);
+    // If no existing data, create an array
+    // Otherwise, convert the localStorage string to an array
+    existing = existing ? existing.split(',') : [];
+    // Get index of announcement id and remove it from the array
+    const index = existing.indexOf(value.toString());
+    if (index > -1) {
+        existing.splice(index, 1);
+    }
+    // Save back to localStorage
+    localStorage.setItem(name, existing.toString());
+};
+
+class ViewHandler extends StateHandler {
+    constructor () {
+        super();
+        this.announcementsHelper = announcementsHelper();
+        this.fetchAdminAnnouncements();
+        this.newTitle = Oskari.getMsg('admin-announcements', 'addNewForm');
+        this.state = {
+            announcements: [],
+            title: this.newTitle,
+            active: true,
+            activeKey: []
+        };
+    }
+
+    fetchAdminAnnouncements () {
+        this.announcementsHelper.getAdminAnnouncements(function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.getAdminAnnouncementsFailed'));
+            } else {
+                this.updateState({
+                    announcements: data.data
+                });
+            }
+        }.bind(this));
+    }
+
+    saveAnnouncement (data) {
+        this.announcementsHelper.saveAnnouncement(data, function (err) {
+            if (err) {
+                Messaging.error(getMessage('messages.saveFailed'));
+            } else {
+                Messaging.success(getMessage('messages.saveSuccess'));
+                this.updateState({
+                    activeKey: []
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+
+    // Update all the announcements f.ex. when saved. Set active key as empty so all panels get closed.
+    updateAnnouncement (data) {
+        this.announcementsHelper.updateAnnouncement(data, function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.updateFailed'));
+                return false;
+            } else {
+                Messaging.success(getMessage('messages.updateSuccess'));
+                removeFromLocalStorageArray(ANNOUNCEMENTS_LOCALSTORAGE, data.id);
+                this.updateState({
+                    activeKey: []
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+
+    deleteAnnouncement (index, id) {
+        this.id = { id };
```

Comment:
Why do you save id as this.id here?

Reply:
Also you could get the index based on the id. Having both means you could end up removing the wrong thing due to some js error. Also it doesn't really make sense to both splice the list locally and fetch a new list entirely. Maybe just do one or the other?